### PR TITLE
Chapter1 homework 3

### PR DESCRIPTION
**Pull request by timk2020**
PR for homework3

### CODE REVIEW

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Checks.cs  1
- Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs  1
- Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367&bid=16135593)

File: Src/BootCamp.Chapter/Checks.cs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Src/BootCamp.Chapter/Program.cs [hunk 2]
```diff
@@ -1,12 +1,14 @@
 ﻿using System;
+using System.Collections.Generic;
+using System.IO;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Src/BootCamp.Chapter/Checks.cs [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+using System.Text;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Src/BootCamp.Chapter/Program.cs [hunk 4]
```diff
@@ -1,12 +1,14 @@
 ﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Program.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs [hunk 5]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
```
Comment (Almantask) [inline Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Src/BootCamp.Chapter/Checks.cs [hunk 6]
```diff
@@ -0,0 +1,41 @@
+﻿using System;
```
Comment (Almantask) [inline Src/BootCamp.Chapter/Checks.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs [hunk 7]
```diff
@@ -0,0 +1,66 @@
+using System;
+using System.IO;
```
Comment (Almantask) [inline Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs [hunk 8]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+using System.Collections.Generic;
```
Comment (Almantask) [inline Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 7
           

Complexity increasing per file
==============================
- Lesson3.cs  1
- Src/BootCamp.Chapter/Lesson3.cs  2
- Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs  1
- Src/BootCamp.Chapter/Checks.cs  1
- Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367&bid=16135593)

File: Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs (hunk 5 above)
Comment (Almantask) [inline Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Lesson3.cs [hunk 9]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
```
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Lesson3.cs [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+
+public class Lesson3
+{
+	public Lesson3()
```
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Replace all tab characters in this file by sequences of white-spaces.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Lesson3.cs [hunk 11]
```diff
@@ -0,0 +1,8 @@
+﻿using System;
+
+public class Lesson3
```
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Move 'Lesson3' into a named namespace.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs (hunk 7 above)
Comment (Almantask) [inline Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Lesson3.cs (hunk 10 above)
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Remove this redundant constructor.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs (hunk 8 above)
Comment (Almantask) [inline Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
- Lesson3.cs  1
- Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs  1
- Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs  1
- Src/BootCamp.Chapter/Checks.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367&bid=16135593)

File: Lesson3.cs (hunk 11 above)
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Move 'Lesson3' into a named namespace.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Lesson3.cs (hunk 9 above)
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Lesson3.cs (hunk 10 above)
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Replace all tab characters in this file by sequences of white-spaces.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Remove this redundant constructor.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Checks.cs  1
- Src/BootCamp.Chapter/Lesson3.cs  2
- Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs  1
- Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs  1
- Lesson3.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367&bid=16135593)

File: Lesson3.cs (hunk 9 above)
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Remove this unnecessary 'using'.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Lesson3.cs (hunk 11 above)
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Move 'Lesson3' into a named namespace.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

File: Lesson3.cs (hunk 10 above)
Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Replace all tab characters in this file by sequences of white-spaces.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

Comment (Almantask) [inline Lesson3.cs]: Codacy Issue found: [Remove this redundant constructor.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
- Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs  1
- Src/BootCamp.Chapter/Checks.cs  1
- Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367&bid=16135593)

**Comment by Almantask** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- Src/BootCamp.Chapter/Lesson3.cs  2
- Tests/BootCamp.Chapter.Tests/Lesson3Tests.cs  1
- Tests/BootCamp.Chapter.Tests/Utils/ConsoleStub.cs  1
- Src/BootCamp.Chapter/Checks.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4865367&bid=16135593)

File: Src/BootCamp.Chapter/Lesson3.cs [hunk 12]
```diff
@@ -0,0 +1,47 @@
+﻿using System;
+namespace BootCamp.Chapter
+{
+    class Lesson3
+    {
+        public static void Demo()
+        {
+            for (int user = 0; user < 2; user++)
+            {
+                string name = ConsolePromptToString("Name: ");
```
Comment (diverges) [inline Src/BootCamp.Chapter/Lesson3.cs]: Consider moving this to its own method

Comment (timk2020) [reply Src/BootCamp.Chapter/Lesson3.cs]: which part do you mean? can you be more specific?

**Pull request closed without merging**